File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
'''
문자열 s 
각 위치마다 앞에 나오고, 자신과 가장 가까운 곳에 있는 같은 글자(뒤에서부터 비교하기)

'''
def solution(s):
    answer = []
    answer.append(-1)
    for i in range(1,len(s)): # 1부터 s까지
        if(s[i] not in s[0:i]):
            answer.append(-1)
        else:
            for j in range(i-1,-1,-1):
                if(s[i] == s[j]):
                    answer.append(i-j)
                    break
                else:
                    continue
        
    return answer

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
        String[] s_array = s.split("");
        int[] answer = new int[s_array.length];
        answer[0] = -1;
        
        for(int i = 1; i < s_array.length; i++) {
        	for(int j = i-1; j > -1 ; j--) {
        		if(s_array[i].equals(s_array[j])) {
        			answer[i] = i-j;
        			break;
        		}
        	}
        }
        
        for(int i = 0; i < answer.length; i++) {
        	if(answer[i] == 0) {
        		answer[i] = -1;
        	}
        }
        
        return answer;
}
}